include/nnstreamer-edge.h

```diff
@@ -96,119 +96,362 @@ typedef void (*nns_edge_data_destroy_cb) (void *data);
 
 /**
  * @brief Create edge handle.
+ * @param[in] id Unique id in local network
+ * @param[in] connect_type value of @a nns_edge_connect_type_e. Connection type between edge nodes.
+ * @param[in] flags value of @a nns_edge_flag_e. The value can be set by using bitwise-or operation.
+ * @param[out] edge_h The edge handle. edge handle should be released using nns_edge_release_handle().
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
+ *
+ * Here is an example of the usage:
+ * @code
+ * // Define edge event callback function.
+ * static int _nns_edge_event_cb (nns_edge_event_h event_h, void *user_data) {
+ *   nns_edge_event_e event_type;
+ *   nns_edge_event_get_type (event_h, &event_type);
+ *
+ *   switch (event_type) {
+ *     case NNS_EDGE_EVENT_CAPABILITY:
+ *     {
+ *       char *capability;
+ *       nns_edge_event_parse_capability (event_h, &capability);
+ *
+ *       // Check capability string and return NNS_EDGE_ERROR_NONE if capability is acceptable.
+ *       // Returning error code will deny the connection.
+ *
+ *       free (capability);
+ *       break;
+ *     }
+ *     case NNS_EDGE_EVENT_NEW_DATA_RECEIVED:
+ *     {
+ *       nns_edge_data_h data_h;
+ *       nns_edge_event_parse_new_data (event_h, &data_h);
+ *
+ *       // You should release returned data_h using nns_edge_data_destroy().
+ *       // Recommend not to spend too much time in event callback.
+ *       nns_edge_data_destroy (data_h);
+ *       break;
+ *     }
+ *     default:
+ *       break;
+ *   }
+ *   return NNS_EDGE_ERROR_NONE;
+ * }
+ *
+ * // Main function
+ * nns_edge_h edge_h;
+ * int ret;
+ *
+ * // create edge handle.
+ * ret = nns_edge_create_handle ("TEMP_ID", NNS_EDGE_CONNECT_TYPE_TCP, NNS_EDGE_FLAG_RECV | NNS_EDGE_FLAG_SEND | NNS_EDGE_FLAG_SERVER, &edge_h);
+ * if (NNS_EDGE_ERROR_NONE != ret)
+ *   // Handle error case. After this, handling error case is repeated, so it is omitted.
+ *
+ * // [Optional] Set edge info if necessary.
+ * ret = nns_edge_set_info (edge_h, "HOST", "127.0.0.1");
+ *
+ * // Set event callback to receive the new data or capability.
+ * // Query-server, query-client and subscriber need to set event callback.
+ * nns_edge_set_event_callback (src->edge_h, _nns_edge_event_cb, src);
```

Comment:
We need cases for:
1. query-server (sink-side & src-side)
2. query-client
3. stream-sink (or pub)
4. stream-src (or sub)

Note that the usage of the term "pub" in query-client is very confusing when we have 3 and 4.